rpc/client.go

```diff
@@ -16,12 +22,21 @@ import (
 )
 
 // UploadReportOptions contains all the required options for uploading a report
-// to both Cedar and the Data Pipes service.
+// to both Cedar and SPS.
+
 type UploadReportOptions struct {
 	Report          *poplar.Report
-	ClientConn      *grpc.ClientConn
+	CedarClientConn *grpc.ClientConn
+	HTTPClient      *http.Client
+	SPSURL          string
 	SerializeUpload bool
 	DryRun          bool
+
+	// SendToCedar specifies whether to send the report to Cedar.
+	SendToCedar bool
+	// SendRatioSPS specifies the probability of sending the report to SPS.
+	// If set to is 1, then the report is always sent to SPS.
+	SendRatioSPS int
```

Comment:
This confuses me a little bit, since I'm not used to seeing ints with probability. It looks like the larger the number, the less likely we are to send the report, is that right?

Reply:
Yes, correct. This behavior is just meant to allow us to sample a small amount of the traffic and make sure we're storing the data correctly with the new system (the original data pipes solution ran into this problem). 

We're likely going to just set this number to like 100 or 50 (1% or 2% of all traffic) and let it run for a day, validate it, and then turn off the Cedar sending, the ratio option, and then send 100% of all traffic to SPS.

Reply:
Merged this just to get started on the Evergreen integration part of this. If changes are need to support sending logs to Splunk, I'll cut another Poplar PR